src/Sets/SparsePolynomialZonotope.jl

```diff
@@ -0,0 +1,314 @@
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+ndependentgens(P::SPZ) = size(P.G, 2)
+
+"""
+    nindependentgens(P::SparsePolynomialZonotope)
+
+Return the number of independent generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+nindependentgens(P::SPZ) = size(P.GI, 2)
+
+"""
+    ngens(P::SparsePolynomialZonotope)
+
+Return the total number of generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Notes
+
+This is equivalent to the number of monomials in the polynomial representation of `P`.
+"""
+ngens(P::SPZ) = ndependentgens(P) + nindependentgens(P)
+
+"""
+    nparams(P::SparsePolynomialZonotope)
+
+Return the number of dependent parameters in the polynomial representation of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The number of dependent parameters in the polynomial representation of P.
+
+### Notes
+
+This corresponds to the number rows in the exponent matrix ``E``.
+"""
+nparams(P::SPZ) = size(P.E, 1)
+
+"""
+    order(P::SparsePolynomialZonotope)
+
+Return the order of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The order of `P`, defined as the quotient between the number of generators and the ambient dimension.
+"""
+order(P::SPZ) = ngens(P) // dim(P)
+
+"""
+    center(P::SparsePolynomialZonotope)
+
+Return the center of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The center of `P`.
+"""
+center(P::SPZ) = P.c
+
+"""
+    dependent_genmat(P::SparsePolynomialZonotope)
+
+Return the matrix of dependent generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+dependent_genmat(P::SPZ) = P.G
+
+"""
+    independent_genmat(P::SparsePolynomialZonotope)
+
+Return the matrix of independent generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+independent_genmat(P::SPZ) = P.GI
+
+
+"""
+    genmat(P::SparsePolynomialZonotope)
+
+Return the full generator matrix of `P`, including dependent and independent generators.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+genmat(P::SPZ) = hcat(P.G, P.GI)
+
+"""
+    expmat(P::SparsePolynomialZonotope)
+
+Return the matrix of exponents of the sparse polynomial zonotope.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The matrix of exponents, where each column is a multidegree.
+
+### Notes
+
+In the exponent matrix, each row corresponds to a parameter (``\alpha_k`` in the definition) and each column to a monomial.
+"""
+expmat(P::SPZ) = P.E
+
+"""
+    indexvector(P::SparsePolynomialZonotope)
+
+Return the index vector of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Notes
+
+The index vector is a vector of positive integers identifing the dependent parameters of `P`.
+"""
+indexvector(P::SPZ) = P.idx
+
+
+"""
+    uniqueID(n::Int)
+
+Returns a collection of n unique identifiers (intergers 1, …, n).
+"""
+uniqueID(n::Int) = 1:n
```

Comment:
This does not really look unique to me...

Reply:
![image](https://user-images.githubusercontent.com/49938764/182086496-8595fc63-e3f7-4b13-9736-5fefdd4e3cf1.png)

indeed to have unique you would need a global counter of how many variables have been created so far and continue always from the new one. shall we implement it that way? The current implementation is kind of recycling the same variables every time, not sure how this would affect the reachability algorithm 🤔 

Anyway, this is probably better addressed and discussed in its own PR (prob shouldn't have added this in the first place, it was just avoid having `1:n` around and having to track those afterwards)

Reply:
Sounds good.